src/opds_dumper.cpp

```diff
@@ -262,6 +262,25 @@ std::string OPDSDumper::categoriesOPDSFeed() const
   );
 }
 
+std::string OPDSDumper::noJSData() const
```

Comment:
This is a interesting point. We could indeed have a `LibraryDumper` (may be only a base class if implementation need it) and have the opds, plain html and libxml dumping using it.
All three dumping are a dumping of the library in different formats. If we can do it only with templates it would be nice, if we cannot ... we cannot. But it would be nice to have the common parts (to be defined) put in common code.

At least, the renaming proposed by @veloman-yunkan seems good.

Reply:
This seems good to me.

(comment based only on the new organisation of  `LibraryDumper`, `OPDSDumper` & `HTMLDumper`. Some other comments may still apply on the code itself)

Reply:
It seems good to me, too.

Can you please rebase and update the history of this PR by extracting `LibraryDumper` out of `OPDSDumper` as the first step and then adding `HTMLDumper`?

Reply:
Done :>